crates/rpc/rpc/src/eth/helpers/sync_listener.rs

```diff
@@ -0,0 +1,130 @@
+//! A utility Future to asynchronously wait until a node has finished syncing.
+
+use futures::Stream;
+use pin_project::pin_project;
+use reth_network_api::NetworkInfo;
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+/// Future that resolves once the node is no longer syncing.
```

Comment:
let's add some more context here, 

This future resolves once the node is no longer syncing: [`NetworkInfo::is_syncing`].